build failure against ghc-8.10: src/TypeFun/Data/List.hs:116:12: error:  Cannot instantiate unification variable ‘b1’

`ghc-8.10` fails to build `type-fun` as:

```
Building library for type-fun-0.1.2..
[5 of 5] Compiling TypeFun.Data.List ( src/TypeFun/Data/List.hs, dist/build/TypeFun/Data/List.o, dist/build/TypeFun/Data/List.dyn_o )

src/TypeFun/Data/List.hs:116:12: error:
    • Cannot instantiate unification variable ‘b1’
      with a type involving polytypes:
        proxy l -> ((l ~ (l :++: '[])) => r) -> r
        GHC doesn't yet support impredicative polymorphism
    • In the expression: unsafeCoerce id
      In an equation for ‘appendId’: appendId = unsafeCoerce id
    • Relevant bindings include
        appendId :: proxy l -> ((l ~ (l :++: '[])) => r) -> r
          (bound at src/TypeFun/Data/List.hs:116:1)
    |
116 | appendId = unsafeCoerce id
    |            ^^^^^^^^^^^^^^^
```

--- a/src/TypeFun/Data/List.hs
+++ b/src/TypeFun/Data/List.hs
@@ -3,6 +3,8 @@
 -- and relies on GHC's type equality (~). The rest of required
 -- operations like 'Reverse' or '(:++:)' you could find in singletons
 
+{-# LANGUAGE GADTs #-}
+
 module TypeFun.Data.List
   ( -- * Primitive operations on lists
     Length
@@ -108,12 +110,16 @@ type family (:++:) (a :: [k]) (b :: [k]) :: [k] where
   '[] :++: b = b
   (a ': as) :++: b = a ': (as :++: b)
 
+data Dict c where Dict :: c => Dict c
+
 appendId
   :: forall proxy l r
    . proxy l
   -> (l ~ (l :++: '[]) => r)
   -> r
-appendId = unsafeCoerce id
+appendId _ r = case obvious of Dict -> r
+  where obvious :: Dict (l ~ (l :++: '[]))
+        obvious = unsafeCoerce (Dict :: Dict (l ~ l))
 
 type IndexOf a s = FromJust (IndexOfMay a s)
 
@@ -172,7 +178,9 @@ subListId
   :: forall proxy l r
    . proxy l
   -> (SubList l l => r) -> r
-subListId _ = unsafeCoerce id
+subListId _ r = case obvious of Dict -> r
+  where obvious :: Dict (SubList l l)
+        obvious = unsafeCoerce (Dict :: Dict ())
 
 type family NotSubList (a :: [k]) (b :: [k]) :: Constraint where
   NotSubList '[]       bs = ()

